image_explorer/image_explorer.py

```diff
@@ -245,6 +247,18 @@ def _get_background(self, xmltree):
             'height': background.get('height')
         })
 
+    def _replace_static_from_url(self, url):
+        if not url:
+            return url
+        try:
+            from static_replace import replace_static_urls
+        except ImportError:
+            return url
+
+        url = '"{}"'.format(url)
+        absolute_url = replace_static_urls(url, course_id=self.course_id)
```

Comment:
The URL returned by `replace_static_urls` isn't absolute, it's relative to the LMS (see [docstring](https://github.com/edx/edx-platform/blob/open-release/ginkgo.1/common/djangoapps/static_replace/__init__.py#L153-L164)).

Is this ok?

E.g., when provided with `<background src="/static/800px-MIT_Dome_night1_Edit.jpg" />`, the returned `student_view_data` contains:

```json
"background":{
  "src":"/assets/courseware/v1/96fc577b3982440ca0087663dbc4552f/asset-v1:OpenCraft+ImageExplorer+2018_1+type@asset+block/800px-MIT_Dome_night1_Edit.jpg",
  "height":null,
  "width":null
}
```

To make it absolute, the mobile app will need to join the URL with the LMS base URL.

The Django way is to use [`request.build_absolute_uri`](https://docs.djangoproject.com/en/1.9/ref/request-response/#django.http.HttpRequest.build_absolute_uri), and I don't think XBlocks have access to the `request` object, so that's not possible here.

Reply:
Please also note this behaviour in the docs for this XBlock's `student_view_data`: https://github.com/edx-solutions/xblock-image-explorer#api-for-native-mobile-frontends